(#3681) Check for correct computer name when refreshing environment variables

corbob opened the pull request:
## Description Of Changes

Update the `EnvironmentHelper` to check for the computername with `$`.

## Motivation and Context

When running as system, the username is actually reported as `COMPUTERNAME$` instead of `COMPUTERNAME`. This is the also the way the check was done when it was written in PowerShell.

## Testing

A new Pester test has been added. The full suite has been run against the current `develop` branch [here](https://sra-soteria-web.ch0.co/buildConfiguration/TestKitchen_AdHoc/36703), which is expected to fail with 2 failed tests. Also ran the full suite of tests against this PR branch [here](https://sra-soteria-web.ch0.co/buildConfiguration/TestKitchen_Chocolatey/36705)

### Operating Systems Testing

- Windows 11 (Manual Tests)
- Windows Server 2016 (Test Kitchen Pester tests)
- Windows Server 2019 (Test Kitchen Pester tests)

## Change Types Made

* [x] Bug fix (non-breaking change).
* [ ] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [ ] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [x] **Developer** Documentation has been updated.
* [x] Tests to cover my changes, have been added.
* [x] All new and existing tests passed?
* [ ] PowerShell code changes: PowerShell v3 compatibility checked?

## Related Issue

Fixes #3681

corbob committed:
(#3681) Check for correct computer name

Previously when we converted this function to C#
(dc409a3de0c03a29d95fec73a555a5d4fedc4ac3) the check for the
computername was inadvertently changed. This commit restores the
comparison to be against the computer name appended with `$`.

corbob commented:
> @corbob can you fix the title of this PR?

Absolutely. Not sure why it picked the branch name as the title, the one I made earlier today picked the commit's title 🤔

vexx32 commented on src/Chocolatey.PowerShell/Helpers/EnvironmentHelper.cs:
```diff
@@ -226,7 +226,8 @@ public static void UpdateSession(PSCmdlet cmdlet)
 
             // User scope should override (be checked after) machine scope, but only if we're not running as SYSTEM
-            if (userName != computerName && userName != Shared.EnvironmentNames.System)
+            // System scope has a username of "COMPUTERNAME$"
+            if (userName != computerName && userName != $"{computerName}$")
             {
```
We... might wanna confirm if the SYSTEM environment name is used like this code originally expected anywhere (older OSes maybe?), maybe we hould be doing all three checks?

corbob committed:
(maint) Chocolatey.PowerShell debugging directions

In order to debug the Chocolatey.PowerShell module in PowerShell, you
need to launch powershell.exe from Visual Studio, and then import the
module into that session. Visual Studio stores the project startup
information in the .csproj.user file which we explicitly ignore as
the user files are for the user's personal settings. In lieu of
committing a user specific file, this commit instead adds a ReadMe to
the Chocolatey.PowerShell project and provides instructions for setting
up for debugging.

corbob committed:
(#3681) Add tests for System Account variables

This commit adds Pester tests to run `refreshenv` as the System account
and ensure that it does not inadvertently pick up the wrong temp
variables.

vexx32 reviewed (approved):
I think I'm happy with this. Good work!
